### PR TITLE
dbeaver/pro#1739 Open external file using preferred application

### PR DESCRIPTION
**Pull request by ShadelessFox**

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by Matvey16** [comment]
Doesn't work on Mac because of file permissions

Screenshot 2023-07-05 at 13 58 22

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/ShellUtils.java [hunk 1]
```diff
@@ -40,6 +43,35 @@ public static boolean launchProgram(@NotNull String path) {
...
+            }
+        } catch (IOException | InterruptedException e) {
+            log.debug("Unable to open external program in a platform-specific way: " + e.getMessage());
+            return false;
+        }
+
+        try {
+            Desktop.getDesktop().open(path.toFile());
+            return true;
+        } catch (IOException e) {
```
Comment (Destrolaric) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/ShellUtils.java]: Linux may not have 'xdg-open' in some cases, we should use try to use 'getDesktop().open()' in that case.

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/ShellUtils.java]: That `return false;` was excessive, I agree.

**Comment by Matvey16** [comment]
Verified on Linux

**Pull request merged**